src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -500,7 +510,26 @@ internal static bool IsFieldOrPropertyInitializer(BoundStatement initializer)
                 {
                     case SyntaxKind.VariableDeclarator:
                     case SyntaxKind.PropertyDeclaration:
-                        return (initializer as BoundExpressionStatement)?.Expression.Kind == BoundKind.AssignmentOperator;
+
+                        switch (initializer.Kind)
+                        {
+                            case BoundKind.Block:
+                                var block = (BoundBlock)initializer;
+                                if (block.Statements.Length == 1)
```

Comment:
Is it possible to have some other number of statements in the block?

Reply:
> Is it possible to have some other number of statements in the block?

In general a bound block can have any number of statements.

---
In reply to: [174217684](https://github.com/dotnet/roslyn/pull/25441#discussion_r174217684) [](ancestors = 174217684)

Reply:
My question is can we reach this case with a different number of statements in the block.
If not, then I think we should assert or throw.

Reply:
> If not, then I think we should assert or throw.

The function was checking if the node had some predefined shape. It was not asserting anything and was not making any assumptions. I am going to keep the same "spirit".

---
In reply to: [174258051](https://github.com/dotnet/roslyn/pull/25441#discussion_r174258051) [](ancestors = 174258051)